assets/javascripts/test_result.js

```diff
@@ -745,40 +745,53 @@ function renderInvestigationTab(response) {
 
     var tbodyElement = document.createElement('tbody');
     Object.keys(response).forEach(key => {
+        var value = response[key];
+        var type = 'pre';
+
+        if (typeof value === 'object' && value.type)
```

Comment:
Repeating my suggestion from #3261 here: might be nice to document the expected formats in a comment.

Reply:
done